monai/inferers/inferer.py

```diff
@@ -122,6 +122,8 @@ class SlidingWindowInferer(Inferer):
             `inputs` and `roi_size`. Output is on the `device`.
         progress: whether to print a tqdm progress bar.
         cache_roi_weight_map: whether to precompute the ROI weight map.
+        cpu_thresh: when provided, adaptively switch to stitching on cpu (to save gpu memory)
```

Comment:
Thanks for the feature enhancement, I would suggest to make this `cpu_thresh` to work with `device=None` only to avoid unexpected behavior.
What do you think?
CC @wyli @rijobro @ericspod 

Thanks.

Reply:
I'm not sure what the concern is? If `device` is any GPU moving the computation to CPU and the results back to it should be fine, so I think the behaviour of the class is that it returns results on the original device regardless. Also it should be state that the `cpu_thresh` value is is number of pixels/voxels, not bytes (which might be a better idea).

Reply:
thanks. I've updated the doc to indicate pixels/voxels. I also think that it should just default to "device" when image is smaller then threshold, we don't need to check for device=None condition

Reply:
agree with Eric that we should use bytes to make it invariant to different dtypes.

Reply:
I think Nic's idea is that if `device="cuda"` the user would expect the output is always on the cuda device. But with `cpu_thresh` the output may quietly go to cpu, slow down the inference/postprocessing.

Reply:
1) I think providing the volume size in bytes will confuse  users. Instead of providing the volume size, they will need to remember to multiply it by 4 (for floats). It's uncommon that the input option for sizes/volumes is in bytes

2) It's true that the output may quietly go to Cpu (but this is intentional to avoid crashes due to GPU memory limit). Currently our inference will fail if at least 1 image in the dataset is abnormally large.  

so what do you guys propose?  to only have this options when device==None (default)?  but this still produces outputs on  both gpu / cpu depending on input image size.  
At the moment it switches to CPU (when image is large), but otherwise respects the user input for "device".

Reply:
so what is the consensus here, is anyone against  specifying the volume in voxels (not in bytes)? if not let's merge it for now

Reply:
Hi @myron ,

I would suggest to make it only work with `device=None` to avoid unexpected or unknown behavior.
If we get more positive user feedback later, we can try to enable the feature for all `device` values.
What do you think?
CC @ericspod @rijobro 

Thanks.

Reply:
Alright, I've changed it, so that the new logics works only if device==None

Reply:
Thanks for your quick update, looks good to me now.